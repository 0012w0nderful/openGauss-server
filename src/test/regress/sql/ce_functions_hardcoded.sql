\! gs_ktool -d all
\! gs_ktool -g

DROP CLIENT MASTER KEY IF EXISTS hardcode_cmk CASCADE;
CREATE CLIENT MASTER KEY hardcode_cmk WITH ( KEY_STORE = gs_ktool , KEY_PATH = "gs_ktool/1" , ALGORITHM = AES_256_CBC);
CREATE COLUMN ENCRYPTION KEY hardcode_cek WITH VALUES (CLIENT_MASTER_KEY = hardcode_cmk, ALGORITHM = AEAD_AES_256_CBC_HMAC_SHA256);

CREATE TABLE IF NOT EXISTS t1(id int, i1 INT ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = hardcode_cek, ENCRYPTION_TYPE = DETERMINISTIC));

INSERT INTO t1 VALUES(1,1),(2,2),(3,3),(4,4),(5,5);

CREATE OR REPLACE FUNCTION select_func() RETURNS INT AS 'SELECT id FROM t1 WHERE i1=1;' LANGUAGE SQL;
SELECT select_func();
CREATE OR REPLACE FUNCTION select_func() RETURNS INT AS 'SELECT id FROM t1 WHERE i1=2' LANGUAGE SQL;
SELECT select_func();
CREATE OR REPLACE FUNCTION select_func() RETURNS INT AS $func_tag$SELECT id FROM t1 WHERE i1=3;$func_tag$ LANGUAGE SQL;
SELECT select_func();
CREATE OR REPLACE FUNCTION select_func() RETURNS INT AS $func_tag$SELECT id FROM t1 WHERE i1=4$func_tag$ LANGUAGE SQL;
SELECT select_func();

CREATE OR REPLACE FUNCTION insert_func() RETURNS VOID AS 'INSERT INTO t1 VALUES(9,9);' LANGUAGE SQL;
SELECT insert_func();
SELECT insert_func();
SELECT insert_func();
SELECT insert_func();

SELECT count(*) from t1 where id=9;

CREATE OR REPLACE FUNCTION insert_select_func() RETURNS SETOF INTEGER AS 'INSERT INTO t1 VALUES(8,8); SELECT id FROM t1 WHERE i1=9;' LANGUAGE SQL;
SELECT insert_select_func();
SELECT insert_select_func();
SELECT insert_select_func();
SELECT insert_select_func();

SELECT count(*) from t1 where id=8;

SELECT * from t1 order by id;

CREATE FUNCTION f_hardcoded_variable() RETURNS int AS $$
BEGIN
RETURN(SELECT id from t1 where i1 = 5 LIMIT 1);
END;
$$ LANGUAGE plpgsql;
SELECT f_hardcoded_variable();
DROP FUNCTION f_hardcoded_variable;
CREATE TABLE t_processed (name text, val INT ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = hardcode_cek, ENCRYPTION_TYPE = DETERMINISTIC), val2 INT);
CREATE FUNCTION f_hardcoded_variable() RETURNS void AS $$
BEGIN
INSERT INTO t_processed (val, val2) VALUES ( 5, 6);
END;
$$ LANGUAGE plpgsql;
CALL f_hardcoded_variable();
DROP FUNCTION f_hardcoded_variable;
SELECT * FROM t_processed ORDER BY name;
DROP TABLE t_processed;
DROP TABLE t1 CASCADE;

create table accounts (
    id serial,
    name varchar(100) not null ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = hardcode_cek, ENCRYPTION_TYPE = DETERMINISTIC),
    balance dec(15,2) not null ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = hardcode_cek, ENCRYPTION_TYPE = DETERMINISTIC),
    primary key(id)
);
CREATE OR REPLACE FUNCTION insert_func100() RETURNS VOID AS 'INSERT INTO accounts(name,balance) VALUES(''100'', 100);' LANGUAGE SQL;
select insert_func100();
CREATE OR REPLACE FUNCTION insert_func200() RETURNS VOID AS $$INSERT INTO accounts(name,balance) VALUES('200', 200);$$ LANGUAGE SQL;
select insert_func200();
CREATE OR REPLACE FUNCTION insert_func300() RETURNS VOID AS $abcd$INSERT INTO accounts(name,balance) VALUES('300', 300);$abcd$ LANGUAGE SQL;
select insert_func300();

select * from accounts ORDER BY id;

DROP TABLE accounts CASCADE;
DROP FUNCTION insert_func100;
DROP FUNCTION insert_func200;
DROP FUNCTION insert_func300;
DROP FUNCTION insert_func;
DROP FUNCTION select_func;
DROP FUNCTION insert_select_func;
DROP COLUMN ENCRYPTION KEY hardcode_cek;
DROP CLIENT MASTER KEY hardcode_cmk;
\! gs_ktool -d all