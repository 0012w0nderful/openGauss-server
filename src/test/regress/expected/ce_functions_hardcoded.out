\! gs_ktool -d all
DELETE ALL

\! gs_ktool -g
GENERATE
1
DROP CLIENT MASTER KEY IF EXISTS hardcode_cmk CASCADE;
NOTICE:  client master key "hardcode_cmk" does not exist
CREATE CLIENT MASTER KEY hardcode_cmk WITH ( KEY_STORE = gs_ktool , KEY_PATH = "gs_ktool/1" , ALGORITHM = AES_256_CBC);
CREATE COLUMN ENCRYPTION KEY hardcode_cek WITH VALUES (CLIENT_MASTER_KEY = hardcode_cmk, ALGORITHM = AEAD_AES_256_CBC_HMAC_SHA256);
CREATE TABLE IF NOT EXISTS t1(id int, i1 INT ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = hardcode_cek, ENCRYPTION_TYPE = DETERMINISTIC));
INSERT INTO t1 VALUES(1,1),(2,2),(3,3),(4,4),(5,5);
CREATE OR REPLACE FUNCTION select_func() RETURNS INT AS 'SELECT id FROM t1 WHERE i1=1;' LANGUAGE SQL;
SELECT select_func();
 select_func 
-------------
           1
(1 row)

CREATE OR REPLACE FUNCTION select_func() RETURNS INT AS 'SELECT id FROM t1 WHERE i1=2' LANGUAGE SQL;
SELECT select_func();
 select_func 
-------------
           2
(1 row)

CREATE OR REPLACE FUNCTION select_func() RETURNS INT AS $func_tag$SELECT id FROM t1 WHERE i1=3;$func_tag$ LANGUAGE SQL;
SELECT select_func();
 select_func 
-------------
           3
(1 row)

CREATE OR REPLACE FUNCTION select_func() RETURNS INT AS $func_tag$SELECT id FROM t1 WHERE i1=4$func_tag$ LANGUAGE SQL;
SELECT select_func();
 select_func 
-------------
           4
(1 row)

CREATE OR REPLACE FUNCTION insert_func() RETURNS VOID AS 'INSERT INTO t1 VALUES(9,9);' LANGUAGE SQL;
SELECT insert_func();
 insert_func 
-------------
 
(1 row)

SELECT insert_func();
 insert_func 
-------------
 
(1 row)

SELECT insert_func();
 insert_func 
-------------
 
(1 row)

SELECT insert_func();
 insert_func 
-------------
 
(1 row)

SELECT count(*) from t1 where id=9;
 count 
-------
     4
(1 row)

CREATE OR REPLACE FUNCTION insert_select_func() RETURNS SETOF INTEGER AS 'INSERT INTO t1 VALUES(8,8); SELECT id FROM t1 WHERE i1=9;' LANGUAGE SQL;
SELECT insert_select_func();
 insert_select_func 
--------------------
                  9
                  9
                  9
                  9
(4 rows)

SELECT insert_select_func();
 insert_select_func 
--------------------
                  9
                  9
                  9
                  9
(4 rows)

SELECT insert_select_func();
 insert_select_func 
--------------------
                  9
                  9
                  9
                  9
(4 rows)

SELECT insert_select_func();
 insert_select_func 
--------------------
                  9
                  9
                  9
                  9
(4 rows)

SELECT count(*) from t1 where id=8;
 count 
-------
     4
(1 row)

SELECT * from t1 order by id;
 id | i1 
----+----
  1 |  1
  2 |  2
  3 |  3
  4 |  4
  5 |  5
  8 |  8
  8 |  8
  8 |  8
  8 |  8
  9 |  9
  9 |  9
  9 |  9
  9 |  9
(13 rows)

CREATE FUNCTION f_hardcoded_variable() RETURNS int AS $$
BEGIN
RETURN(SELECT id from t1 where i1 = 5 LIMIT 1);
END;
$$ LANGUAGE plpgsql;
SELECT f_hardcoded_variable();
 f_hardcoded_variable 
----------------------
                    5
(1 row)

DROP FUNCTION f_hardcoded_variable;
CREATE TABLE t_processed (name text, val INT ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = hardcode_cek, ENCRYPTION_TYPE = DETERMINISTIC), val2 INT);
CREATE FUNCTION f_hardcoded_variable() RETURNS void AS $$
BEGIN
INSERT INTO t_processed (val, val2) VALUES ( 5, 6);
END;
$$ LANGUAGE plpgsql;
CALL f_hardcoded_variable();
 f_hardcoded_variable 
----------------------
 
(1 row)

DROP FUNCTION f_hardcoded_variable;
SELECT * FROM t_processed ORDER BY name;
 name | val | val2 
------+-----+------
      |   5 |    6
(1 row)

DROP TABLE t_processed;
DROP TABLE t1 CASCADE;
create table accounts (
    id serial,
    name varchar(100) not null ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = hardcode_cek, ENCRYPTION_TYPE = DETERMINISTIC),
    balance dec(15,2) not null ENCRYPTED WITH (COLUMN_ENCRYPTION_KEY = hardcode_cek, ENCRYPTION_TYPE = DETERMINISTIC),
    primary key(id)
);
NOTICE:  CREATE TABLE will create implicit sequence "accounts_id_seq" for serial column "accounts.id"
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "accounts_pkey" for table "accounts"
CREATE OR REPLACE FUNCTION insert_func100() RETURNS VOID AS 'INSERT INTO accounts(name,balance) VALUES(''100'', 100);' LANGUAGE SQL;
select insert_func100();
 insert_func100 
----------------
 
(1 row)

CREATE OR REPLACE FUNCTION insert_func200() RETURNS VOID AS $$INSERT INTO accounts(name,balance) VALUES('200', 200);$$ LANGUAGE SQL;
select insert_func200();
 insert_func200 
----------------
 
(1 row)

CREATE OR REPLACE FUNCTION insert_func300() RETURNS VOID AS $abcd$INSERT INTO accounts(name,balance) VALUES('300', 300);$abcd$ LANGUAGE SQL;
select insert_func300();
 insert_func300 
----------------
 
(1 row)

select * from accounts ORDER BY id;
 id | name | balance 
----+------+---------
  1 | 100  |  100.00
  2 | 200  |  200.00
  3 | 300  |  300.00
(3 rows)

DROP TABLE accounts CASCADE;
DROP FUNCTION insert_func100;
DROP FUNCTION insert_func200;
DROP FUNCTION insert_func300;
DROP FUNCTION insert_func;
DROP FUNCTION select_func;
DROP FUNCTION insert_select_func;
DROP COLUMN ENCRYPTION KEY hardcode_cek;
DROP CLIENT MASTER KEY hardcode_cmk;
\! gs_ktool -d all
DELETE ALL
 1
